src/export-plan/pdf-gen.ts

```diff
@@ -0,0 +1,328 @@
+import jsPDF from 'jspdf';
+import autoTable from 'jspdf-autotable';
+import {
+  getCollegeFullName,
+  getMajorFullName,
+  getMinorFullName,
+  sortedSemesters,
+  isPlaceholderCourse,
+  getGradFullName,
+} from '../utilities';
+import store from '../store';
+import { addFonts } from './add-fonts';
+import { getCollegeAbbr } from '@/requirements/data/index';
+
+const rowHeight = 18;
+const tableWidth = 516;
+// vertical space between tables for two semesters
+const tableGap = 20;
+const firstTableY = 170;
+
+const rowFontSize = 10.5;
+const headerFontSize = 10.5;
+
+const reqsToFilterOut = ['A&S Credits'];
+
+const bubbleColourMap: Record<RequirementGroupType, (req?: string) => string> = {
+  College: (req?: string) =>
+    req && store.state.userRequirementsMap[req].sourceSpecificName === 'UNI'
+      ? '#1AA9A5'
+      : '#4F7D91',
+  Grad: () => '#1E8481',
+  Major: () => '#1E8481',
+  Minor: () => '#145351',
```

Comment:
Hm, switch required a default for some reason, which I thought looked a bit less clean, since this is a complete set of cases for the data type. And I was getting a linter warning about it wanting a "return" at the end, even though I was returning in every case.

switch/case syntax in typescript can't be used as an expression I think, so it just makes things somewhat ugly. Like I can't do = return switch () {
...}

Reply:
I think a switch statement is still preferred. In other areas of the codebase we either throw an error or return a value that is never used.

Reply:
Hey, I switched the other map to switch/case but I think it is cleanest to keep this one like this. It's used in two places, and for one use case we have a requirement with a requirement group, and for the other case we just want to get the colour for a group like "grad", and this format works best.

Reply:
alright